`test_manual_teardown` runs into timeouts

See the failure here, for example:

https://github.com/LiberTEM/LiberTEM/actions/runs/5532215051/job/14981114471

```
_____________________________ test_manual_teardown _____________________________

    def test_manual_teardown():
        p = Process(target=_teardown_manual, name="_teardown_manual")
        p.start()
        p.join(30)
        exitcode = p.exitcode
        if exitcode != 0:
            p.terminate()
>           raise RuntimeError(f"exitcode is {exitcode}, should be 0")
E           RuntimeError: exitcode is None, should be 0

tests/executor/test_pipelined.py:518: RuntimeError
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,17 +23,9 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: Choose Python version ${{ matrix.python_version }}
-        if: ${{ matrix.os != 'macos-latest' || matrix.python_version != '3.7' }}
         uses: actions/setup-python@v4
         with:
           python-version: '${{ matrix.python_version }}'
-          cache: 'pip'
-      # workaround for https://github.com/actions/setup-python/issues/682
-      - name: Choose Python version 3.7.16 for macos-latest
-        if: ${{ matrix.os == 'macos-latest' && matrix.python_version == '3.7' }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.7.16'
           cache: 'pip'
       - run: |
            [[ "${{ matrix.os }}" = "ubuntu-latest" ]] && sudo apt-get install -y libcap-dev || true

--- a/setup.py
+++ b/setup.py
@@ -167,7 +167,7 @@ setup(
         'opentelemetry-api',
         'tblib',
         'tomli',
-        'sparseconverter>=0.3.2',
+        'sparseconverter>=0.3.3',
     ],
     extras_require={
         # NumPy interfacing issue on Win 11, Python 3.10

